dotnet/test/common/BiDi/Network/NetworkEventsTest.cs

```diff
@@ -132,11 +132,8 @@ public async Task CanListenToFetchError()
 
         await using var subscription = await context.Network.OnFetchErrorAsync(tcs.SetResult);
 
-        try
-        {
-            await context.NavigateAsync("https://not_a_valid_url.test", new() { Wait = ReadinessState.Complete });
-        }
-        catch (Exception) { }
+        var navigateTask = context.NavigateAsync("https://not_a_valid_url.test", new() { Wait = ReadinessState.Complete });
+        await ((Task)navigateTask).ConfigureAwait(ConfigureAwaitOptions.SuppressThrowing);
```

Comment:
Do we think it is better than `try/catch` block in our case?

@YevgeniyShunevych your feedback is appreciated.

Reply:
Actually I meant this pattern in my comment: https://github.com/SeleniumHQ/selenium/pull/14776#discussion_r1849016071

Reply:
Honestly, I don't like very much both old and new approaches. What if we use `Assert.That` (which also handles well async functions) here too:

```cs
Assert.That(
    async () => context.NavigateAsync("https://not_a_valid_url.test", new() { Wait = ReadinessState.Complete }),
    Throws.TypeOf<SomeException>()); // Or just Throws.Exception if we don't care about exception type.
```

Maybe "assertion" was not intended to be at that place. But with exception assertion (instead of swallowing) we are confident that a method actually thrown something.

Reply:
Otherwise I can propose to create a helper method "wait and swallow exception". This way it will be clearer in tests. And that exception swallowing logic will be encapsulated in one place. There I would probably use try/catch inside.

Reply:
OK, let me decide: @RenderMichael please don't introduce the pattern `await ... SuppressThrowing` in this PR - this is really another topic for discussion. For the `Assert.That(..., Throws...)` yes and yes, kind of road to NUnit 4. Thank you!

Reply:
> Otherwise I can propose to create a helper method "wait and swallow exception". This way it will be clearer in tests. And that exception swallowing logic will be encapsulated in one place.

That is basically what we are doing with `ConfigureAwait(SuppressThrowing)`.

> this is really another topic for discussion

Good point. I'll remove it from this PR. I can open a separate issue (or a PR directly if it's known to be desirable?) about migrating to NUnit 4.

Reply:
OK, I removed `SuppressThrowing` from the actual tests. There are two types which are used as part of test infrastructure: `RemoteSeleniumServer` and `TestWebServer`. These are actually the only places which were important to me to use this: the `SuppressThrowing` option avoid first-chance exceptions and simplify the debugging flow.

Here are the diffs:

![image](https://github.com/user-attachments/assets/7784fbd6-2dab-434e-b8bb-1e2228a72ab7)
![image](https://github.com/user-attachments/assets/d159fe81-a06d-4cb5-8f38-53ea91f7db14)
![image](https://github.com/user-attachments/assets/8a9b7859-1b0f-421e-85d3-e9917f495105)

If we prefer to stay away from this format in the test infrastructure as well, that's OK by me.

Reply:
I can easily accept the original pattern you proposed in this PR, but I cannot accept the `await ...SuppressThrowing`. If we want to merge this PR, then let's make changes aligned.

Reply:
Reverted